internal/storage/minio_chunk_manager.go

```diff
@@ -286,7 +286,16 @@ func (mcm *MinioChunkManager) MultiRemove(keys []string) error {
 // RemoveWithPrefix removes all objects with the same prefix @prefix from minio.
 func (mcm *MinioChunkManager) RemoveWithPrefix(prefix string) error {
 	objects := mcm.Client.ListObjects(mcm.ctx, mcm.bucketName, minio.ListObjectsOptions{Prefix: prefix, Recursive: true})
-	for rErr := range mcm.Client.RemoveObjects(mcm.ctx, mcm.bucketName, objects, minio.RemoveObjectsOptions{GovernanceBypass: true}) {
+	mode, _, _, err := mcm.Client.GetBucketObjectLockConfig(mcm.ctx, mcm.bucketName)
```

Comment:
We can delete GovernanceBypass config directly,  it should not be a backward compatibility issue.

Reply:
Is there any user already using S3 with object lock? I don't know. It's your calls anyway.

Reply:
It's better to keep this object lock check.

Reply:
we will not consider backward compatibility because the data should be granted modification privilege, or it is against data process flow within Milvus.

Reply:
enable object means both compaction and delete will fail.

Reply:
Updated